exes/mqtt-broadcaster/inc/mqtt_broadcaster.hpp

```diff
@@ -0,0 +1,618 @@
+
+    auto p_id = mqtt_client_->acquire_unique_packet_id();
+
+    logger_.trace("Preparing subscription packet for topic: {}", ncmd_topic);
+
+    auto subscribe_packet = async_mqtt::v5::subscribe_packet{
+      *p_id, { { async_mqtt::allocate_buffer(ncmd_topic), async_mqtt::qos::at_most_once | async_mqtt::sub::nl::yes } }
+    };
+
+    logger_.trace("Sending subscription packet...");
+
+    co_await mqtt_client_->send(subscribe_packet, asio::use_awaitable);
+
+    logger_.trace("Subscription packet sent. Waiting for SUBACK...");
+
+    co_await mqtt_client_->recv(async_mqtt::filter::match, { async_mqtt::control_packet_type::suback }, asio::use_awaitable);
+  }
+
+  auto add_new_signals() -> void {
+    logger_.trace("Starting to add new signals...");
+
+    ipc_client_.signals([this](const std::vector<tfc::ipc_ruler::signal>& signals) -> void {
+      logger_.trace("Received {} new signals to add.", signals.size());
+
+      signals_.clear();
+      signals_.reserve(signals.size());
+      for (auto signal : signals) {
+        // slot must include type name
+        std::string slot_name;
+        switch (signal.type) {
+          case details::type_e::_bool:
+            slot_name = fmt::format("bool_slot_mqtt_broadcaster_{}", signal.name);
+            break;
+          case details::type_e::unknown:
+            slot_name = fmt::format("unknown_slot_mqtt_broadcaster_{}", signal.name);
+            break;
+          case details::type_e::_int64_t:
+            slot_name = fmt::format("int64_t_slot_mqtt_broadcaster_{}", signal.name);
+            break;
+          case details::type_e::_uint64_t:
+            slot_name = fmt::format("uint64_t_slot_mqtt_broadcaster_{}", signal.name);
+            break;
+          case details::type_e::_double_t:
+            slot_name = fmt::format("double_t_slot_mqtt_broadcaster_{}", signal.name);
+            break;
+          case details::type_e::_string:
+            slot_name = fmt::format("string_slot_mqtt_broadcaster_{}", signal.name);
+            break;
+          case details::type_e::_json:
+            slot_name = fmt::format("json_slot_mqtt_broadcaster_{}", signal.name);
+            break;
+        }
+
+        auto ipc = tfc::ipc::details::create_ipc_recv<tfc::ipc::details::any_recv>(io_ctx_, slot_name);
+
+        std::visit(
+            [&](auto&& receiver) -> void {
+              using receiver_t = std::remove_cvref_t<decltype(receiver)>;
+              if constexpr (!std::same_as<receiver_t, std::monostate>) {
+                auto error_code = receiver->connect(signal.name);
+                if (error_code) {
+                  logger_.trace("Error connecting to signal: {}, error: {}", signal.name, error_code.message());
+                }
+              }
+            },
+            ipc);
+
+        signal_data signal_data{ signal, std::move(ipc), std::nullopt };
+
+        signals_.emplace_back(std::move(signal_data));
```

Comment:
you can emplace straight away and save the boiler plate.
```suggestion
        signals_.emplace_back(signal, std::move(ipc), std::nullopt);
```